### ISSUE TITLE
Collect provenance inside if and for statements

### ISSUE THREAD

**Issue by blernermhc** [End-to-end-provenance/RDataTracker#110]
Currently, if and for statements are treated as an individual statement.  As a result, our provenance data cannot tell simple things like how many times a loop is iterated, or which branch of an if statement is followed.

We should also make sure that there are not other control constructs with similar issues.

**Comment by blernermhc** [comment]
There are also while and repeat loops, as well as blocks.  

switch is a function rather than a statement in R and I don't think needs any special attention.

Here is a script that could be used to test how we handle control constructs.
```
{
  a <- 1
  b <- 2
}

if (b == 1) {
  cc <- 3
} else {
  d <- 4
}

repeat {
  d <- d + 1
  if (d == 7) {
    break
  }
}

while (d < 10) {
  d <- d + 1
}

for (e in 1:5) {
  f <- e + 1
}

j <- switch (b, g = 7, h = 8, i = 9)

```

**Comment by erboose** [comment]
The feature branch control annotates a script to collect provenance for statements inside if, for, repeat, while, and simple block structures.  Internal statements are wrapped with ddg.eval.  ddg.start and ddg.finish are added to blocks in for, repeat, and while structures.

The resulting ddgs are close but not quite correct.  The current plan is to refactor and simplify .ddg.parse.commands and ddg.eval before returning to complete this task.

We may also want to give some thought to limiting the number of nodes created for statements that involve a large number of repetitions (e.g. a for loop with 10000 iterations).

**Comment by MKLau** [comment]
This is now being addressed by Issue #181
